modules/ocf_prometheus/manifests/server.pp

```diff
@@ -0,0 +1,152 @@
+class ocf_prometheus::server {
+  include apache
+  include apache::mod::proxy
+  include apache::mod::proxy_http
+  include ocf::firewall::allow_web
+  include ocf::ssl::default
+
+  file {
+    '/usr/local/bin/gen-prometheus-nodes':
+      source => 'puppet:///modules/ocf_prometheus/gen-prometheus-nodes',
+      mode   => '0755';
+  }
+
+  cron { 'gen-prometheus-nodes':
+    command => '/usr/local/bin/gen-prometheus-nodes /var/local/prometheus-nodes.json',
+    minute  => '0',
+    require => File['/usr/local/bin/gen-prometheus-nodes'];
+  }
+
+  exec { 'gen-promethues-nodes-initial':
+    command => '/usr/local/bin/gen-prometheus-nodes /var/local/prometheus-nodes.json',
+    creates => '/var/local/prometheus-nodes.json',
+  }
+
+  file {
+    '/usr/local/bin/gen-prometheus-printers':
+      source => 'puppet:///modules/ocf_prometheus/gen-prometheus-printers',
+      mode   => '0755';
+  }
+
+  cron { 'gen-prometheus-printers':
+    command => '/usr/local/bin/gen-prometheus-printers /var/local/prometheus-printers.json',
+    minute  => '0',
+    require => File['/usr/local/bin/gen-prometheus-printers'];
+  }
+
+  exec { 'gen-prometheus-printers-initial':
+    command => '/usr/local/bin/gen-prometheus-printers /var/local/prometheus-printers.json',
+    creates => '/var/local/prometheus-printers.json',
```

Comment:
this task seems common enough that a better pattern than this would exist for it - is there a way we can consolidate these `exec`s in a nicer way?

Reply:
Make a defined resource?

Reply:
Shouldn't the cron job have a dependency on the initial script running first?

Reply:
also why does the cron job require the `/usr/local/bin/gen-prometheus-printers` but not the `exec` command?

Reply:
Something like this seems a lot more sane. I disagree that a `defined` type is necessary unless this pattern repeats in even more places. So far two places isn't enough to warrant the abstraction.

```
file {
    '/usr/local/bin/gen-prometheus-printers':
      source => 'puppet:///modules/ocf_prometheus/gen-prometheus-printers',
      mode   => '0755';
  } ->
 exec { 'gen-prometheus-printers-initial':
    command => '/usr/local/bin/gen-prometheus-printers /var/local/prometheus-printers.json',
    creates => '/var/local/prometheus-printers.json',
  } ->
 cron { 'gen-prometheus-printers':
    command => '/usr/local/bin/gen-prometheus-printers /var/local/prometheus-printers.json',
    minute  => '0',
  }
```

Reply:
If this pattern ever repeats a third time (which may actually happen in the future) I'd be willing to look into ways to cut down on the repetition. But for now I'll just take keur's suggestion of using the arrows.